cupy/_core/dlpack.pyx

```diff
@@ -320,34 +448,34 @@ cdef inline _ndarray_base _dlpack_to_cupy_array(dltensor) except +:
     else:
         raise TypeError('Unsupported dtype. dtype code: {}'.format(dtype.code))
 
-    mem_ptr = memory.MemoryPointer(mem, mem.dlm_tensor.dl_tensor.byte_offset)
-    cdef int64_t ndim = mem.dlm_tensor.dl_tensor.ndim
+    mem_ptr = memory.MemoryPointer(mem, mem.dl_tensor.byte_offset)
+    cdef int64_t ndim = mem.dl_tensor.ndim
 
-    cdef int64_t* shape = mem.dlm_tensor.dl_tensor.shape
+    cdef int64_t* shape = mem.dl_tensor.shape
     cdef vector[Py_ssize_t] shape_vec
     shape_vec.assign(shape, shape + ndim)
 
-    if mem.dlm_tensor.dl_tensor.strides is NULL:
-        # Make sure this capsule will never be used again.
-        cpython.PyCapsule_SetName(mem.dltensor, 'used_dltensor')
+    if mem.dl_tensor.strides is NULL:
         return core.ndarray(shape_vec, cp_dtype, mem_ptr, strides=None)
-    cdef int64_t* strides = mem.dlm_tensor.dl_tensor.strides
+    cdef int64_t* strides = mem.dl_tensor.strides
     cdef vector[Py_ssize_t] strides_vec
     for i in range(ndim):
         strides_vec.push_back(strides[i] * (bits // 8))
 
-    # Make sure this capsule will never be used again.
-    cpython.PyCapsule_SetName(mem.dltensor, 'used_dltensor')
     return core.ndarray(shape_vec, cp_dtype, mem_ptr, strides=strides_vec)
 
 
-cpdef from_dlpack(array):
+def from_dlpack(array, *, device=None, copy=None):
```

Comment:
Note: Changing `cpdef` to `def` would cut out the Cython exposure. I don't suppose any users to use this from Cython, so I am fine with it. Just wanna call it out.